arch/x86/core/intel64/cache_s.S

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2013-2014 Wind River Systems, Inc.
+ * Copyright (c) 2021 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/**
+ * @file
+ * @brief Cache manipulation
+ *
+ * This module contains functions for manipulating caches.
+ */
+
+#include <toolchain.h>
+
+#ifndef CONFIG_CLFLUSH_INSTRUCTION_SUPPORTED
+
+#if defined(CONFIG_CLFLUSH_DETECT)
+
+	#define CACHE_FLUSH_NAME z_cache_flush_wbinvd
+	#define CPUID_CFLSH_BIT (1 << 19)
+
+.global z_is_clflush_available
+z_is_clflush_available:
+	pushq %rbx
+	movq $1, %rax
+	cpuid
+	movq %rdx, %rax
+	andq $CPUID_CFLSH_BIT, %rax
+	popq %rbx
+	ret
```

Comment:
Mild complaint: runtime feature detection isn't really a Zephyr thing.  Obviously x86 is right at the boundary where sometimes you have to, but in this case?  I dunno, cache line size is 512 bytes across the whole universe now and CLFLUSH works everywhere too.  I honestly think you could skip this, and if we really need it across architectures we have in-tree it belongs in devicetree instead (and I say that as someone who really hates devicetree and loves x86 assembly!)

Reply:
Ok that means updating ia32 as well then (I just c/p cache_s.S from there with minor modifications)